lib/fluent/plugin/out_forward.rb

```diff
@@ -346,6 +350,31 @@ def stop
       end
     end
 
+    def before_shutdown
+      super
+      @suspend_flush = true
+    end
+
+    def after_shutdown
+      last_ack if @require_ack_response
+      super
+    end
+
+    def try_flush
+      return if @require_ack_response && @suspend_flush
+      super
+    end
+
+    def last_ack
+      overwrite_delayed_commit_timeout
+      select_interval = if @delayed_commit_timeout > 3
+                          1
+                        else
+                          @delayed_commit_timeout / 3.0
+                        end
```

Comment:
Same code exists in `ack_reader`. Could you unify them?

Reply:
@ashie We unified them.